server/internal/core/instance/service.go

```diff
@@ -56,3 +67,33 @@ type InitialAdmin struct {
 	Username string
 	Password string
 }
+
+func (s *Service) Status(ctx context.Context) (Status, error) {
+	completed, err := s.FirstTimeSetupCompleted(ctx)
+	if err != nil {
+		return Status{}, err
+	}
+
+	return Status{
+		Info:                    s.info,
+		FirstTimeSetupCompleted: completed,
+	}, nil
+}
+
+type StatusDB string
+type StatusFileStorage string
+
+const (
+	StatusDBPostgres StatusDB = "postgres"
+	StatusDBSqlite   StatusDB = "sqlite"
+)
+
+const (
+	StatusFileStorageFS StatusFileStorage = "filesystem"
+	StatusFileStorageS3 StatusFileStorage = "s3"
+)
```

Comment:
❓ Aren't we mixing infra and core?

Reply:
The instance context is a bit meta in nature, as it represents Oxidrive itself. Here we're not using the mentioned infrastructure components, they are just labels to be shown to users.

We could move these constants outside the core, but it doesn't really change the code or coupling.
The point of separating domain and infrastructure is to avoid coupling, simplify testing by allowing to inject mocks, and make it easier to swap implementations. These strings do not introduce any coupling, so I say it's fine

Reply:
I don't see a reason for moving them outside, we have only two values for each enum. If we add more implementations we can think about a better solution. For now it's ok for me.